### DISCUSSION TITLE
Incorrect chinese tokenization with zh_core_web_md model when chinese characture is beside with english or unicode symbol

### DISCUSSION THREAD

**Discussion by brightgems** [explosion/spaCy#6538] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

when I use spacy to toenize chiense web text with zh_core_web_md model, the tokenized word is not accurate when chinese characture is beside with english or unicode symbol

below is exmple code to reproduce:

```
nlp=spacy.load('zh_core_web_md')
nlp.tokenizer.pkuseg_update_user_dict(["修颜", "太大"])
doc = nlp("★★★倩碧VC安瓶平替这几年维生素C的风也是挺大的,就普普通通的保湿面霜。")
for span in doc.sents:
    sentence = [doc[i] for i in range(span.start, span.end)]
    print(sentence)
``

my output is 
```
[★★, ★倩]
[碧V, C安瓶, 平替, 这几年, 维生素C, 的, 风, 也, 是, 挺, 大, 的, ,, 就, 普普通通, 的, 保湿, 面霜, 。]
```

## Your Environment
<!-- Include details of your environment. If you're using spaCy 1.7+, you can also type `python -m spacy info --markdown` and copy-paste the result here.-->
* Operating System: Ubuntu
* Python Version Used: 3.7
* spaCy Version Used: 2.3
* Environment Information:

**Comment by adrianeboyd** [comment, marked as answer]
Hi, the segmentation model is a statistical model trained on OntoNotes, where it's possible this kind of sequence only occurred rarely or maybe even not at all in the training data, so the model doesn't handle it well.

You can fine-tune an existing model with your own data, see "Example 8" in the `pkuseg` docs: https://github.com/lancopku/pkuseg-python/blob/master/readme/readme_english.md#usage

A note for the future: if you're fine-tuning the models for spacy v3.0+, you'll want to `import spacy_pkuseg as pkuseg` so that you can load our slightly modified model format (which just uses `msgpack` instead of `pickle` to avoid the security issues around `pickle`).

**Reply by brightgems** [reply]
＠adrianeboyd

Thanks very much for your instruction. Generally, I understand a customized tokenization model should bed trained for my corpus. 

Given that this step is ready, I might still want to take use of the other pipelines like tag/dependency parse. I guess the new pkuseg model file should  copy to zh_core_web_md/tokenizer/pkuseg_model folder and override existing file, so that only tokenizer are replace while other component can still work.

please correct me if anything is wrong. THANKS!

**Reply by adrianeboyd** [reply]
Yes, that sounds right. The `pkuseg_model` folder is a standard pkuseg model that can be loaded/saved separately with pkuseg directly.

If the word segmentation changes drastically with your custom pkuseg model, the existing components won't work correctly anymore, but if you're just fine-tuning the model a little, you should be able to drop in a replacement and keep the existing components without any problems.
